libpromises/locks.c

```diff
@@ -554,6 +554,35 @@ static void PromiseTypeString(char *dst, size_t dst_size, const Promise *pp)
     }
 }
 
+// returns true if pid points to a cfengine process
+static bool IsCfengineProcess(const int pid)
+{
+    // There is no /proc on windows
+#ifndef _WIN32
+    char* procfile = xcalloc(1024, sizeof(char));
+    char* cmd = xcalloc(1024, sizeof(char));
```

Comment:
Either `free()` these variables or, better, use arrays like `char procfile[PATH_MAX]` and similar.